modules/extension/app-schema/app-schema/src/main/java/org/geotools/data/complex/AppSchemaDataAccess.java

```diff
@@ -93,6 +93,14 @@
     private FilterFactory2 filterFac = CommonFactoryFinder.getFilterFactory2(null);
 
     /**
+     * Flag to mark non-accessible data accesses, which should be automatically
+     * disposed of when no longer needed by any accessible data access.
+     */
+    boolean hidden = false;
+
+    List<Name> relatedTypeNames = new ArrayList<>();
```

Comment:
@ridethepenguin this new member relatedTypeNames is not used anywhere and should be removed.